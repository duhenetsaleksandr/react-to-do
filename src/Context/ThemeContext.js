import React from 'react';

const ThemeContext = React.createContext({ darkTheme: true, switchTheme: () => {} });

export default ThemeContext;
