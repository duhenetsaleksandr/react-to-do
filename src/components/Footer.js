import React from 'react';
import '../styles/footer.scss';
import Social from './Social';
import Copyright from './Copyright';

export default function Footer() {
    return (
        <footer className="footer">
            <div className="container footer__container">
                <Copyright/>
                <Social/>
            </div>
        </footer>
    );
}
