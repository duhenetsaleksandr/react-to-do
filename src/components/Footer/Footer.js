import React, { useContext } from 'react';
import ThemeContext from 'Context/ThemeContext';
import './Footer.scss';
import Social from 'components/Social/Social';
import Copyright from 'components/Copyright';

const Footer = () => {
    const { darkTheme } = useContext(ThemeContext);
    return (
        <footer className={`footer ${darkTheme ? 'dark' : 'light'}`}>
            <div className="container footer__container">
                <Copyright/>
                <Social/>
            </div>
        </footer>
    );
};

export default Footer;
