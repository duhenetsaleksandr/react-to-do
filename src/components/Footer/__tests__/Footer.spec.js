import React from 'react';
import Footer from '../Footer';
import ThemeContext from 'Context/ThemeContext';

describe('should render Footer component', () => {
    let component;
    beforeEach(() => {
        component = mount(<Footer/>);
    });

    it('should contain tag footer with class .footer', () => {
        const footer = component.find('footer.footer');
        expect(footer).toHaveLength(1);
    });

    it('should contain div with class .container and .footer__container', () => {
        const footer = component.find('div.container.footer__container');
        expect(footer).toHaveLength(1);
    });

    it('should contain component Copyright', () => {
        const copyright = component.find('Copyright');
        expect(copyright).toHaveLength(1);
    });

    it('should contain component Copyright', () => {
        const social = component.find('Social');
        expect(social).toHaveLength(1);
    });

    it('should render Footer with light theme using context', () => {
        const component = mount(<Footer/>, {
            wrappingComponent: ThemeContext.Provider,
            wrappingComponentProps: {
                value: { darkTheme: false },
            },
        });

        const footer = component.find('footer.footer.light');
        expect(footer).toHaveLength(1);
    });

    it('should render Footer with dark theme using context', () => {
        const component = mount(<Footer/>, {
            wrappingComponent: ThemeContext.Provider,
            wrappingComponentProps: {
                value: { darkTheme: true },
            },
        });

        const footer = component.find('footer.footer.dark');
        expect(footer).toHaveLength(1);
    });

    it('should render Footer component snapshot', () => {
        const component = shallow(<Footer/>);
        expect(component).toMatchSnapshot();
    });
});
