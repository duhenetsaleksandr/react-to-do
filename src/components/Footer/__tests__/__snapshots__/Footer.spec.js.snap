// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Footer component should render Footer component snapshot 1`] = `
<footer
  className="footer dark"
>
  <div
    className="container footer__container"
  >
    <Copyright />
    <Social />
  </div>
</footer>
`;
